### PR TITLE
CDK: Fix typing errors

### LINKED ISSUE #8576: CDK: fix license headers and mypy errors
## Current Behavior
We use pre-commit to validate python code locally, currently, CDK does not pass pre-commit checks, which makes it less usable if usable at all.

## Expected Behavior
The code in CDK is properly formatted and pre-commit checks are passed.

## Logs
```
base) ➜  python git:(keu/cdk-base-pydantic-models) ✗ pre-commit run --files airbyte_cdk/sources/config.py
licenseheaders...........................................................Passed
black....................................................................Passed
isort....................................................................Passed
prettier.............................................(no files to check)Skipped
flake8...................................................................Passed
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

airbyte-cdk/python/airbyte_cdk/sources/config.py: note: In member "schema" of class "BaseConfig":
airbyte-cdk/python/airbyte_cdk/sources/config.py:60:9: error: Returning Any
from function declared to return "Dict[str, Any]"  [no-any-return]
            return schema
            ^
airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py:29: note: In module imported here,
airbyte-cdk/python/airbyte_cdk/sources/__init__.py:1: note: ... from here:
airbyte-cdk/python/airbyte_cdk/sources/utils/schema_helpers.py:13:1: error:
Module 'airbyte_cdk.models' does not explicitly export attribute
'ConnectorSpecification'; implicit reexport disabled  [attr-defined]
    from airbyte_cdk.models import ConnectorSpecification
    ^
airbyte-cdk/python/airbyte_cdk/sources/utils/schema_helpers.py: note: In member "__call__" of class "JsonFileLoader":
airbyte-cdk/python/airbyte_cdk/sources/utils/schema_helpers.py:32:9: error:
Returning Any from function declared to return "Dict[str, Any]"  [no-any-return]
            return json.load(open(uri))
            ^
airbyte-cdk/python/airbyte_cdk/sources/utils/schema_helpers.py: note: In function "resolve_ref_links":
airbyte-cdk/python/airbyte_cdk/sources/utils/schema_helpers.py:47:9: error:
Returning Any from function declared to return "Dict[str, Any]"  [no-any-return]
            return obj
            ^
```

### PR DESCRIPTION
**Pull request by keu**
## What
closes #8576 

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>

#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: airbyte-cdk/python/unit_tests/sources/streams/http/test_http.py [hunk 1]
```diff
@@ -406,14 +398,14 @@ class CacheHttpStreamWithSlices(CacheHttpStream):
...
-        return f'{stream_slice.get("path")}'
+        return f'{stream_slice["path"]}' if stream_slice else ""
 
     def stream_slices(self, **kwargs) -> Iterable[Optional[Mapping[str, Any]]]:
         for path in self.paths:
             yield {"path": path}
 
     def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
-        yield response
+        yield {"value": len(response.text)}
```
Comment (eliziario) [inline airbyte-cdk/python/unit_tests/sources/streams/http/test_http.py:408]: Just out of curiosity, why this radical change?

**Review by eliziario — APPROVED** [review]
nice to me, have just one non-critical question

File: airbyte-cdk/python/unit_tests/sources/streams/http/test_http.py (hunk 1 above)
Comment (keu) [reply airbyte-cdk/python/unit_tests/sources/streams/http/test_http.py:408]: it needs to be a mapping according to signature

File: airbyte-cdk/python/unit_tests/sources/test_abstract_source.py [hunk 2]
```diff
@@ -108,7 +108,7 @@ def test_discover(mocker):
     assert expected == src.discover(logger, {})
 
 
-def test_read_nonexistent_stream_raises_exception(mocker, logger):
```
Comment (augan-rymkhan) [inline airbyte-cdk/python/unit_tests/sources/test_abstract_source.py:115]: Fixture "logger" not found  [error](https://github.com/airbytehq/airbyte/runs/4799300904?check_suite_focus=true#step:5:1161)
I removed logger fixture from the tests, so they will use logger defined in the same module.

**Review by vitaliizazmic — APPROVED** [review]

**Pull request merged**